src/main/java/org/jenkinsci/plugins/workflow/graph/StandardGraphLookupView.java

```diff
@@ -24,50 +25,62 @@
 
     static final String INCOMPLETE = "";
 
-    /** Map the blockStartNode to its endNode, to accellerate a range of operations */
+    /** Map the blockStartNode to its endNode, to accelerate a range of operations */
     HashMap<String, String> blockStartToEnd = new HashMap<>();
 
     /** Map a node to its nearest enclosing block */
     HashMap<String, String> nearestEnclosingBlock = new HashMap<>();
 
+    final ReentrantLock lock = new ReentrantLock();
+
     public void clearCache() {
-        blockStartToEnd.clear();
-        nearestEnclosingBlock.clear();
+        lock.lock();
+        try {
+            blockStartToEnd.clear();
+            nearestEnclosingBlock.clear();
+        } finally {
+            lock.unlock();
```

Comment:
https://bugs.openjdk.java.net/browse/JDK-7084550 :cry: Could also just use a simple monitor, since you are not using any special features of `ReentrantLock`.

Reply:
For that matter, the simplest fix would probably just be to slap `synchronized` on all these methods. The only caller is https://github.com/jenkinsci/workflow-api-plugin/blob/01fbb0814a68950f927e9ee0739b2c6f835a2daa/src/main/java/org/jenkinsci/plugins/workflow/flow/FlowExecution.java#L256-L318 so there is no risk of using `this` as the monitor.